functions/src/api/v1/group/createGroup.ts

```diff
@@ -47,12 +63,39 @@ export const createGroup = wrap(async (req: Request, res: Response) => {
     await assertNewResourceCreationGranted(userId)
 
     const name = req.body.name
+    const requestedGroupId = req.body.groupId
+    const password = req.body.password
     const members = JSONParse(req.body.members || '[]')
 
-    const groupId = await GroupDao.add(userId, name, members)
-    await GroupDao.update({
-        id: groupId,
-    })
+    if (!password) {
+        throw new BadRequestError('Password is required')
+    }
+
+    let groupId: GroupId
+    if (requestedGroupId) {
+        try {
+            await GroupDao.get(requestedGroupId)
+            throw new GroupConflictError()
+        } catch (error) {
+            if (error instanceof GroupNotFoundError) {
+                await GroupDao.set(
+                    requestedGroupId,
+                    userId,
+                    name,
+                    password,
+                    members
+                )
+                groupId = requestedGroupId
+            } else {
+                throw error
+            }
+        }
+    } else {
+        groupId = await GroupDao.add(userId, name, password, members)
+        await GroupDao.update({
```

Comment:
why doesn't `groupDao.add` set the `groupId` as well ?

Reply:
the "add" add a new object within a collection, letting firestore pick the right id so we dont have any conflicted id.

Reply:
(non blocking) but why calling `GroupDao.update` ? Isn't firebase id part of the entity, after the call to `add` ?

Reply:
the document id is not within the document itself on add, and is retrieve differently where quering the whole collection. It make things simpler to we know the id is always within the document.